library/core/src/char/methods.rs

```diff
@@ -1822,9 +1822,10 @@ pub const fn encode_utf8_raw(code: u32, dst: &mut [u8]) -> &mut [u8] {
 /// Panics if the buffer is not large enough.
 /// A buffer of length 2 is large enough to encode any `char`.
 #[unstable(feature = "char_internals", reason = "exposed only for libstd", issue = "none")]
-#[rustc_const_unstable(feature = "const_char_encode_utf16", issue = "130660")]
+#[rustc_const_stable(feature = "const_char_encode_utf16", since = "CURRENT_RUSTC_VERSION")]
 #[doc(hidden)]
 #[inline]
+#[rustc_allow_const_fn_unstable(const_eval_select)]
 pub const fn encode_utf16_raw(mut code: u32, dst: &mut [u16]) -> &mut [u16] {
```

Comment:
The way this works changed in https://github.com/rust-lang/rust/pull/131349. I believe `encode_utf16_raw` needs to lose the `rustc_const_{un}stable` and `rustc_allow_const_fn_unstable` gates and instead get `#[cfg_attr(not(bootstrap), rustc_const_stable_indirect)]`. 

You might also need to change the `rustc_const_stable` on `encode_utf16` to `#[cfg_attr(not(bootstrap), rustc_const_stable)]` so it only becomes const stable when the new stability checking is available.

Cc @rust-lang/wg-const-eval, I'm not positive about the right thing to do while the new checks aren't in bootstrap.

Reply:
I hope the docs at https://rustc-dev-guide.rust-lang.org/stability.html#rustc_const_unstable are helpful :)

Reply:
But yes, an `unstable` function cannot be `rustc_const_stable` in the new regime. But the trouble is this code needs to build both with the bootstrap compiler and the latest nightly compiler, so the result should be something like
```rust
#[cfg_attr(bootstrap, rustc_const_stable(feature = "const_char_encode_utf16", since = "CURRENT_RUSTC_VERSION"))]
#[rustc_allow_const_fn_unstable(const_eval_select)]
```
Also please add a comment explaining why allowing const_eval_select is okay.

> You might also need to change the rustc_const_stable on encode_utf16 to #[cfg_attr(not(bootstrap), rustc_const_stable)] so it only becomes const stable when the new stability checking is available.

That should not be needed.